### ISSUE TITLE
Lint/CopDirectiveSyntax flags directive text inside `--` reasons

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15636]
## Description:

`Lint/CopDirectiveSyntax` treats directive-looking text after a valid `--` reason as a second directive.
That reports a malformed directive offense even though the text after `--` is only explanatory text.

## Expected behavior:

`Lint/CopDirectiveSyntax` should ignore directive-looking text inside a valid `--` reason comment.

This comment should not register an offense:

```ruby
# rubocop:disable Layout/LineLength -- see # rubocop:disable Style/Encoding in old branch
```

## Actual behavior:

`Lint/CopDirectiveSyntax` scans the whole comment text for directives, including the text after `--`.

As a result, it treats the directive mentioned in the reason as a second directive on the same line and reports:

```text
Malformed directive comment detected. Only the first directive on a line takes effect. List the cop names in a single directive instead.
```

## Steps to reproduce the problem:

Create `directive_reason.rb`:

```ruby
# rubocop:disable Layout/LineLength -- see # rubocop:disable Style/Encoding in old branch
```

Run:

```console
bundle exec ruby exe/rubocop directive_reason.rb --only Lint/CopDirectiveSyntax -f simple
```

Current `master` reports one offense, but the text after `--` is plain reason text and should not be parsed as another directive.

## RuboCop version:

```console
1.90.0 (using Parser 3.3.12.0, rubocop-ast 1.50.0, analyzing as Ruby 2.7, running on ruby 3.3.11) +Rubydex [x86_64-linux]
  - rubocop-performance 1.27.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```

**Comment by RedZapdos123** [comment]
Opened a PR to resolve the issue.
